install_config/persistent_storage/persistent_storage_flex_volume.adoc

```diff
@@ -0,0 +1,304 @@
+masters and nodes and will be used in subsequent `detach` call as
+`<volume-name>`. Any `'/'` character in the `<volume-name>` are automatically
+replaced by `'~'`.
+* Arguments: `<json>`
+* Executed on: master, node
+* Expected output: default JSON + `volumeName`:
++
+[source,json]
+----
+{
+	"status": "Success",
+	"message": "",
+	"volumeName": "foo-volume-bar" <1>
+}
+----
+<1> Unique name of the volume in storage backend foo.
+
+`attach`::
+* Attaches a volume represented by the JSON to given node. This operation
+should return name of the device on the node if it is known at this time
+(i.e. it has been assigned by the storage backend at this point).
+If the device is not known at this time, the device must be found on the
+node by subsequent `waitforattach` operation.
+* Arguments: `<json>` `<node-name>`
+* Executed on: master
+* Expected output: default JSON + `device` (if known):
++
+[source,json]
+----
+{
+	"status": "Success",
+	"message": "",
+	"device": "/dev/xvda" <1>
+}
+----
+<1> Name of the device on the node (if known).
+
+`waitforattach`::
+* Waits until a volume is fully attached to a node and its device emerges.
+If previous `attach` operation has returned `<device-name>`, it is provided
+as an input parameter, otherwise `<device-name>` is empty and it's task of
+this operation to find the device on the node.
+* Arguments: `<device-name>` `<json>`
+* Executed on: node
+* Expected output: default JSON + `device`
++
+[source,json]
+----
+{
+	"status": "Success",
+	"message": "",
+	"device": "/dev/xvda" <1>
+}
+----
+<1> Name of the device on the node.
+
+`detach`::
+* Detaches given volume from a node. `<volume-name>` is name of the device as
+returned by `getvolumename` operation. Note that `<volume-name>` has all
+`'/'` characters replaced with `'~'`!
+* Arguments: `<volume-name>` `<node-name>`
+* Executed on: master
+* Expected output: default JSON
+
+
+`isattached`::
+* Checks that a volume is attached to a node.
+* Arguments: `<json>` `<node-name>`
+* Executed on: master
+* Expected output: default JSON + `attached`
++
+[source,json]
+----
+{
+	"status": "Success",
+	"message": "",
+	"attached": true <1>
+}
+----
+<1> Status of attachment of the volume to the node.
+
+`mountdevice`::
+* Mounts a volume's device to a directory. `<device-name>` is name of the
+device as returned by previous `waitforattach` operation.
+* Arguments: `<mount-dir>` `<device-name>` `<json>`
+* Executed on: node
+* Expected output: default JSON
+
+`unmountdevice`::
+* Unmounts a volume's device from a directory.
+* Arguments: `<mount-dir>`
+* Executed on: node
+
+All other operations should return JSON with `{"status": "Not supported"}`
+and exit code `1`.
+
+[NOTE]
+====
+Master-initiated attach/detach operations are enabled by default in
+{product-title} 3.6. They may work in older versions, but must be explicitly
+enabled, see
+xref:enabling_controller_attach_detach.adoc#[Enabling Controller-managed Attachment and Detachment].
```

Comment:
xref:../install_config/persistent_storage/enabling_controller_attach_detach.adoc#enabling-controller-attach-detach[Enabling Controller-managed Attachment and Detachment]